ultrasonic/src/main/kotlin/org/moire/ultrasonic/service/DownloadFile.kt

```diff
@@ -0,0 +1,392 @@
+                    if (save) {
+                        Util.renameFile(partialFile, saveFile)
+                        mediaStoreService.saveInMediaStore(this@DownloadFile)
+                    } else {
+                        Util.renameFile(partialFile, completeFile)
+                    }
+                }
+            } catch (x: Exception) {
+                Util.close(outputStream)
+                Util.delete(completeFile)
+                Util.delete(saveFile)
+                if (!isCancelled) {
+                    isFailed = true
+                    Timber.w(x, "Failed to download '%s'.", song)
+                }
+            } finally {
+                Util.close(inputStream)
+                Util.close(outputStream)
+                if (wakeLock != null) {
+                    wakeLock.release()
+                    Timber.i("Released wake lock %s", wakeLock)
+                }
+                wifiLock?.release()
+                CacheCleaner(context).cleanSpace()
+                downloader.value.checkDownloads()
+            }
+        }
+
+        private fun acquireWakeLock(wakeLock: WakeLock?): WakeLock? {
+            var wakeLock1 = wakeLock
+            if (Util.isScreenLitOnDownload(context)) {
+                val pm = context.getSystemService(Context.POWER_SERVICE) as PowerManager
+                val flags = PowerManager.SCREEN_DIM_WAKE_LOCK or PowerManager.ON_AFTER_RELEASE
+                wakeLock1 = pm.newWakeLock(flags, toString())
+                wakeLock1.acquire(10 * 60 * 1000L /*10 minutes*/)
+                Timber.i("Acquired wake lock %s", wakeLock1)
+            }
+            return wakeLock1
+        }
+
+        override fun toString(): String {
+            return String.format("DownloadTask (%s)", song)
+        }
+
+        private fun downloadAndSaveCoverArt(musicService: MusicService) {
+            try {
+                if (!TextUtils.isEmpty(song.coverArt)) {
+                    val size = Util.getMinDisplayMetric(context)
+                    musicService.getCoverArt(context, song, size, true, true)
+                }
+            } catch (x: Exception) {
+                Timber.e(x, "Failed to get cover art.")
+            }
+        }
+
+        @Throws(IOException::class)
+        fun InputStream.copyTo(out: OutputStream, onCopy: (totalBytesCopied: Long) -> Any): Long {
+            var bytesCopied: Long = 0
+            val buffer = ByteArray(DEFAULT_BUFFER_SIZE)
+            var bytes = read(buffer)
+            while (!isCancelled && bytes >= 0) {
+                out.write(buffer, 0, bytes)
+                bytesCopied += bytes
+                onCopy(bytesCopied)
+                bytes = read(buffer)
+            }
+            return bytesCopied
+        }
+    }
+
+    private fun setProgress(totalBytesCopied: Long) {
+        if (song.size != null) {
+            progress.postValue((totalBytesCopied * 100 / song.size!!).toInt())
+        }
+    }
+
+    companion object {
```

Comment:
The converter likes to do this with all static functions, but IMHO in kotlin for private functions the companion object just adds bloat and is unnecessary.

Reply:
Ok done :)